_clients/OSC-dot-net.md

```diff
@@ -0,0 +1,329 @@
+```
+
+## Searching for a document
+
+To search for a student indexed above, you want to construct a query that is analogous to the following Query DSL query:
+
+```json
+GET students/_search
+{
+  "query" : {
+    "match": {
+      "lastName": "Santos"
+    }
+  }
+}
+```
+
+The query above is a shorthand version of the following explicit query:
+
+```json
+GET students/_search
+{
+  "query" : {
+    "match": {
+      "lastName": {
+        "query": "Santos"
+      }
+    }
+  }
+}
+```
+
+In OpenSearch.Client, this query looks like this:
+
+```cs
+var searchResponse = client.Search<Student>(s => s
+                                .Index("students")
+                                .Query(q => q
+                                    .Match(m => m
+                                        .Field(fld => fld.LastName)
+                                        .Query("Santos"))));
+```
+
+You can print out the results by accessing the documents in the response:
+
+```cs
+if (searchResponse.IsValid)
+{
+    foreach (var s in searchResponse.Documents)
+    {
+        Console.WriteLine($"{s.Id} {s.LastName} {s.FirstName} {s.Gpa} {s.GradYear}");
+    }
+}
+```
+
+The response contains one document, which corresponds to the correct student:
+
+`100 Santos Paulo 3.93 2021`
+
+## Using OpenSearch.Client methods asynchronously
+
+For applications that require asynchronous code, all method calls in OpenSearch.Client have asynchronous counterparts:
+
+```cs
+// synchronous method
+var response = client.Index(student, i => i.Index("students"));
+
+// asynchronous method
+var response = await client.IndexAsync(student, i => i.Index("students"));
+```
+
+## Falling back on the low-level OpenSearch.Net client
+
+OpenSearch.Client exposes the low-level OpenSearch.Net client you can use if anything is missing:
+
+```cs
+var lowLevelClient = client.LowLevel;
+
+var searchResponseLow = lowLevelClient.Search<SearchResponse<Student>>("students",
+    PostData.Serializable(
+        new
+        {
+            query = new
+            {
+                match = new
+                {
+                    lastName = new
+                    {
+                        query = "Santos"
+                    }
+                }
+            }
+        }));
+
+if (searchResponseLow.IsValid)
+{
+    foreach (var s in searchResponseLow.Documents)
+    {
+        Console.WriteLine($"{s.Id} {s.LastName} {s.FirstName} {s.Gpa} {s.GradYear}");
+    }
+}
+```
+
+## Sample program
+
+The following is a complete sample program that illustrates all of the concepts above. It uses the Student class described above.
+
+```cs
+using OpenSearch.Client;
+using OpenSearch.Net;
+
+namespace NetClientProgram;
+
+internal class Program
+{
+    private static IOpenSearchClient osClient = new OpenSearchClient();
+
+    public static void Main(string[] args)
+    {       
+        Console.WriteLine("Indexing one student......");
+        var student = new Student { Id = 100, 
```

Comment:
When I copy this program into my IDE I get an error "The type or namespace name 'Student' could not be found (are you missing a using directive or an assembly reference?)"

Do we need to define student somewhere? ex. 
```
public class Student
{
   ....
}
```

Reply:
Yes, you should copy the class Student (defined at the beginning of the file) to the same project - in a separate file from your program